### PR TITLE
Controll delen till rapport

### PR DESCRIPTION
**Pull request by Scitt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,69 @@ The navigation system comprised SLAM provided by the Cartographer online node an
 The global plan was generated by the default global planner known simply as global_planner.
 </p>
 
+### Control
```
Comment (zachtyll) [inline README.md:108]: Run the text through Grammarly to remove minor spelling mistakes.

File: README.md [hunk 2]
```diff
@@ -105,6 +105,69 @@ The navigation system comprised SLAM provided by the Cartographer online node an
...
+
+![](https://github.com/D7039E-E7032E/Project-Report/blob/Controll-delen-till-rapport/images/Control_Str.svg)
+
+In the illustration above the structure of the control system an its supporting components. 
+
+#### Global Navigator
+
+The purpose of the global navigator is to find a path the robot should follow to reach its goal. The the goal points is set manually and when the robot reaches it, a new one will be chosen by the operator. It will use information from the global map, a map of the combined scans from all robots. This is implemented using the already existing ROS package move_base that calculates a global path for the robot using a form of A-star algorithm. 
+
+The Global Navigator uses maps generated from cartographer, which is a slow process, combined with A-star not being a particular fast algorithm. This makes the Global Navigator a rather slow. This is solved with a Local Navigator.
```
Comment (zachtyll) [inline README.md:120]: Change this section. We are making use of Dijkstras algorithm to create a global path plan. This is more resource intense than using a faster A* algorithm.

File: README.md [hunk 3]
```diff
@@ -105,6 +105,69 @@ The navigation system comprised SLAM provided by the Cartographer online node an
...
+
+#### Local Navigator
+
+The Local Navigator's purpose is to avoid obstacles and keeping an optimal distance to walls, this to give optimal scanning quality. It will follow the direction of the path given by the layout navigator.
```
Comment (zachtyll) [inline README.md]: Is layout navigator used to reference the global navigator?

File: README.md [hunk 4]
```diff
@@ -105,6 +105,69 @@ The navigation system comprised SLAM provided by the Cartographer online node an
...
+
+![Force](https://github.com/D7039E-E7032E/Project-Report/blob/Controll-delen-till-rapport/images/Force.svg)
+
+A custom ROS node was implemented to solve this. The node operates by giving all point seen by the lidar a virtual force, see equation above, which when combined creates an heat map. The node then solves the gradient of the heat map in the position of the robot, as in equation below. The value of *lp* is the distance to the point, and the values *c* and *n* is tuning-parameters, which was tuned manually. The resulting vector will then point in a direction away from all obstacles.
```
Comment (zachtyll) [inline README.md]: use *l<sub>p</sub>* sub-tags for describing index *p* of *l*. sup-tags gives superscript output.

**Review by zachtyll — CHANGES_REQUESTED** [review]

**Review by anti002 — DISMISSED** [review]
When the requested changes from zachtyll are resolved this should be fine

**Review by carfri — DISMISSED** [review]

File: README.md (hunk 4 above)
Comment (Scitt) [reply README.md]: vet inte hur man gör detta, du får göra det.

Comment (oscrim) [reply README.md]: man gör som i html, l<sub>p</sub> = `l<sub>p</sub>`

File: README.md [hunk 5]
```diff
@@ -115,13 +115,13 @@ In the illustration above the structure of the control system an its supporting
 
 #### Global Navigator
 
-The purpose of the global navigator is to find a path the robot should follow to reach its goal. The the goal points is set manually and when the robot reaches it, a new one will be chosen by the operator. It will use information from the global map, a map of the combined scans from all robots. This is implemented using the already existing ROS package move_base that calculates a global path for the robot using a form of A-star algorithm. 
+The purpose of the global navigator is to find a path the robot should follow to reach its goal. The the goal points are set manually and when the robot reaches it, a new one will be chosen by the operator. It will use information from the global map, a map of the combined scans from all robots. This was implemented using the already existing ROS package move_base that calculates a global path for the robot using Dynamic Window Approach Planner algorithm.
```
Comment (zachtyll) [inline README.md:118]: Dynamic Window Approach Planner is for Local Navigation while Global_Planner is for global plans. Global_Planner uses Dijkstras algorithm.

**Review by zachtyll — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (zachtyll) [reply README.md:118]: Actually it seems I am wrong about this. It is the /move_base/DWAPlannerROS/global_plan that is used and not the /move_base/NavfnROS/plan.

**Review by zachtyll — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (Scitt) [reply README.md:118]: du får fixa detta själv då så att blir rätt

**Review by zachtyll — APPROVED** [review]

**Review by anti002 — APPROVED** [review]

**Pull request merged**